File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root:
            return []
        q=[]
        result=[]
        q.append(root)
        while len(q)>0:
            levelList=[]
            l=len(q)
            for i in range(l):
                r=q.pop(0)
                levelList.append(r.val)
                if r.left is not None:
                    q.append(r.left)
                if r.right is not None:
                    q.append(r.right)
            result.append(levelList)
        return result

        

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private List<List<Integer>> BFS(TreeNode root){
        if(root==null){
            return new ArrayList<>();
        }
        int i,l;
        Queue<TreeNode> q=new LinkedList<>();
        List<List<Integer>> result=new ArrayList<>();
        q.add(root);
        while(!q.isEmpty()){
            List<Integer> levelList=new ArrayList<>();
            l=q.size();
            for(i=0;i<l;i++){
                TreeNode r=q.poll();
                levelList.add(r.val);
                if(r.left!=null){
                    q.add(r.left);
                }
                if(r.right!=null){
                    q.add(r.right);
                }
            }
            result.add(levelList);
        }
        return result;
    }
    public List<List<Integer>> levelOrder(TreeNode root) {
        return BFS(root);
    }
}